src/Internal/Marshaller/Type/DateIntervalType.php

```diff
@@ -53,10 +53,10 @@ public function serialize($value): int
         $method = 'total' . \ucfirst($this->format);
 
         if ($this->format === DateInterval::FORMAT_NANOSECONDS) {
-            return DateInterval::parse($value, $this->format)->totalMicroseconds * 1000;
+            return (int)(DateInterval::parse($value, $this->format)->totalMicroseconds * 1000);
```

Comment:
Or maybe int64 overflow just happens? In this case, the conversion to the float occurs...

Reply:
Can't be the case, the test case where I got it was under 10000ms.